Source/WebKit/Platform/IPC/Attachment.h

```diff
@@ -48,7 +49,11 @@ namespace IPC {
 class Decoder;
 class Encoder;
 
+#if OS(DARWIN)
```

Comment:
1. Typically inheritance for composition purposes causes unforeseen problems down the line.
2. At least it's not nice to be ifdefing the types.
3. The code with he type variant and such don't make sense (doesn't make much sense afterwards, but..)


I think this should be as simple to write as m_sendRight, but you need to add the.` release() `method ? eg. now you call `attachment.leakSendRight() `and after you would call `attachment.release().leakSendRight() `?